src/main/scala/beam/agentsim/infrastructure/parking/ParkingZoneFileUtils.scala

```diff
@@ -488,37 +466,33 @@ object ParkingZoneFileUtils extends ExponentialLazyLogging {
     }
   }
 
+  private def calculateNumStalls(
+    initialNumStalls: Double
+  )(implicit parkingStallCountScalingFactor: Double, rand: Random): Int = {
+    val expectedNumberOfStalls = initialNumStalls * parkingStallCountScalingFactor
+    val floorNumberOfStalls = math.floor(expectedNumberOfStalls).toInt
+    if ((expectedNumberOfStalls % 1.0) > rand.nextDouble) {
+      floorNumberOfStalls + 1
+    } else {
+      floorNumberOfStalls
+    }
```

Comment:
We could use beam.utils.MathUtils#roundUniformly for the same purpose. Though in this case a new overloaded version of this method is required: the one that gets an instance of Random as an input parameter.